Repository/ClassRepository.cs

```diff
@@ -0,0 +1,53 @@
+﻿using mPole.Data.DbContext;
+using mPole.Data.Models;
+using mPole.Interface.Repositories;
+using Microsoft.EntityFrameworkCore;
+
+namespace mPole.Repository
+{
+    public class ClassRepository : IClassRepository
+    {
+        private readonly ApplicationDbContext _context;
+
+        public ClassRepository(ApplicationDbContext context)
+        {
+            _context = context;
+        }
+
+        public async Task<Class> GetClassByIdAsync(int poleDanceClassId, CancellationToken cancellationToken)
+        {
+            var poleDanceClass = await _context.Classes
+                                .Where(t => t.Id == poleDanceClassId)
+                                .Include(t => t.Training)
+                                .Include(t => t.RegisteredUsers)
+                                .AsNoTracking()
+                                .FirstOrDefaultAsync();
+
+            if (poleDanceClass == null)
+            {
+                throw new ArgumentNullException($"Class with id {poleDanceClassId} not found");
+            }
+
+            return poleDanceClass;
+        }
+
+        public async Task<ICollection<Class>> GetClassesByDateRangeAsync(DateTime startDate, DateTime endDate, CancellationToken cancellationToken)
+        {
+            var classes = await _context.Classes
+                .Where(c => c.Date >= startDate && c.Date <= endDate)
+                .AsNoTracking()
+                .Include(c => c.Training)
+                .Include(c => c.RegisteredUsers)
+                .ToListAsync(cancellationToken);
```

Comment:
Hmm to jest dłuższy temat, sam z tym w Workaiu się spotkałem - nadużywanie ToList/ToListAsync - domena młodych programistów C# - warto rozważyć w niektórych przypadkach - nie mówię ze w tym konkretnym, ale ogólnie trzymanie w pobieraniach kolekcji danych aby jak najdłużej było IQueryable bazodanowe i według wzorców materializować poprzez ToList lub AsEnumerable powinniśmy na etapie serwisu w logice biznesowej - ale to temat do przemyślenia na po obronie - bo przy Twojej ilości danych appka to ssie w milisekundach, ale gdy faktyczni userzy np 2000 userów zacznie korzystać, to będziesz bekał tym ToList-em jak ja w Learningu ;)

Reply:
super, dodam na liste TO DOs